delft/hydra-proxy.nix

```diff
@@ -1,71 +1,35 @@
 { config, lib, pkgs, ...}:
 
-let
-
-  hydraProxyConfig =
-    ''
-      TimeOut 900
-
-      <Proxy *>
-        Order deny,allow
-        Allow from all
-      </Proxy>
-
-      ProxyRequests     Off
-      ProxyPreserveHost On
-      ProxyPass         /apache-errors !
-      ProxyPass         /.well-known !
-      ErrorDocument 503 /apache-errors/503.html
-      ProxyPass         /       http://127.0.0.1:3000/ retry=5 disablereuse=on
-      ProxyPassReverse  /       http://127.0.0.1:3000/
-
-      <Location />
-        SetOutputFilter DEFLATE
-        BrowserMatch ^Mozilla/4\.0[678] no-gzip\
-        BrowserMatch \bMSI[E] !no-gzip !gzip-only-text/html
-        SetEnvIfNoCase Request_URI \.(?:gif|jpe?g|png|narinfo)$ no-gzip dont-vary
-        SetEnvIfNoCase Request_URI /api/ no-gzip dont-vary
-        SetEnvIfNoCase Request_URI /download/ no-gzip dont-vary
-        SetEnvIfNoCase Request_URI /nar/ no-gzip dont-vary
-      </Location>
-    '';
-
-in
-
 {
-
-  services.httpd = {
+  services.nginx = {
     enable = true;
-    adminAddr = "edolstra@gmail.com";
-    logFormat = ''"%h %l %u %t \"%r\" %>s %b %D"'';
-    extraConfig = hydraProxyConfig +
-      ''
-        RewriteEngine On
-        RewriteCond %{HTTPS} off
-        RewriteRule (.*) https://%{HTTP_HOST}%{REQUEST_URI} [R=301,L]
+    enableReload = true;
+
+    recommendedBrotliSettings = true;
+    recommendedGzipSettings = true;
+    recommendedOptimisation = true;
+    recommendedProxySettings = true;
+    recommendedTlsSettings = true;
+    recommendedZstdSettings = true;
+
+    proxyTimeout = "900s";
+
+    virtualHosts."hydra.nixos.org" = {
+      forceSSL = true;
+      enableACME = true;
+
+      extraConfig = ''
+        error_page 503 /503.html;
+        location = /503.html {
+          root ${./nginx-error-pages};
+          internal;
+        }
       '';
 
-    virtualHosts."hydra.nixos.org" =
-      { forceSSL = true;
-        enableACME = true;
-        extraConfig = ''
-          # Required by Catalyst.
-          RequestHeader set X-Forwarded-Proto https
-          RequestHeader set X-Forwarded-Port 443
-          Header always set Strict-Transport-Security "max-age=15552000"
-
-          RewriteEngine On
-          RewriteCond %{QUERY_STRING} ^query=pkgs.chromium$
```

Comment:
Is there still a chromium wiki page that looks to a rather expensive hydra search which causes lots of traffic?

Reply:
2 requests in the last 14 days, not worth caring about.

Reply:
> Is there still a chromium wiki page that looks to a rather expensive hydra search which causes lots of traffic?

The page was switched to use search.nixos.org
https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/linux/chromium_packages.md